resolwe/flow/models.py

```diff
@@ -535,7 +572,23 @@ def save(self, render_name=False, *args, **kwargs):
 
             # We can only save dependencies after the data object has been saved. This
             # is why a transaction block is needed and the save method must be called first.
-            self.save_dependencies(self.input, self.process.input_schema)  # pylint: disable=no-member
+            if create:
+                self.save_dependencies(self.input, self.process.input_schema)  # pylint: disable=no-member
+
+        if create:
+            self.create_entity()
+
+    def delete(self, *args, **kwargs):
+        """Delete Data object and its Entity if it becomes empty."""
```

Comment:
I am not sure if it is ok to do this by overriding `delete`. In bulk deletes, it can happen that this method is not called. Use the `post_delete` signal instead, which is called even for bulk deletes.